src/launcher.rs

```diff
@@ -881,4 +925,44 @@ mod tests {
         let error = Launcher::new(Some(V2.clone())).unwrap_err().to_string();
         assert_eq!(error, "the requested version (2.0.0) is not installed");
     }
+
+    #[test]
+    fn should_run_existing_shutdown_script() {
+        let _ = logging::init();
+
+        let script_path = Path::new(SHUTDOWN_SCRIPT_PATH);
+        let output_path = Path::new(SHUTDOWN_SCRIPT_OUTPUT_PATH);
+        if script_path.exists() {
+            fs::remove_file(script_path).expect("Couldn't remove existing test shutdown script");
+        }
+        if output_path.exists() {
+            fs::remove_file(output_path)
+                .expect("Couldn't remove existing test shutdown script output");
+        }
+        let script_contents = format!(
+            "#!/bin/bash\necho \"{}\" > {} && exit {}",
+            SHUTDOWN_SCRIPT_SUCCESS_OUTPUT, SHUTDOWN_SCRIPT_OUTPUT_PATH, SHUTDOWN_SCRIPT_EXIT_CODE
```

Comment:
We should probably assert if we really exit with `SHUTDOWN_SCRIPT_EXIT_CODE`. Maybe the code behind the `test` feature [here](https://github.com/casper-network/casper-node-launcher/pull/42/files#diff-329ff24ea49112211e6e62c9af6a3f4744826f8bcece2add5cf216f93fb01752R434) could set some lazy global flag which we can read here in the test. Or maybe it is possible to read the launcher logs here and assert that the expected message is there?

We should also cover the case when we return `0` as a result of the script being non-existent.

Reply:
I tried to do the lazy global flag thing but it ended up pretty ugly, so I scrapped it. What we really need is an integration test outside the launcher, with all the right conditions. I could try to create another mock node, see how that goes.